src/features/transactions/components/app-call-transaction-info.tsx

```diff
@@ -21,92 +27,133 @@ const foreignApplicationsTabId = 'foreign-applications'
-        children: <ApplicationArgs transaction={transaction} />,
-      },
-      {
-        id: foreignAccountsTabId,
-        label: foreignAccountsTabLabel,
-        children: <ForeignAccounts transaction={transaction} />,
-      },
-      {
-        id: foreignApplicationsTabId,
-        label: foreignApplicationsTabLabel,
-        children: <ForeignApplications transaction={transaction} />,
-      },
-      {
-        id: foreignAssetsTabId,
-        label: foreignAssetsTabLabel,
-        children: <ForeignAssets transaction={transaction} />,
-      },
-      {
-        id: globalStateDeltaTabId,
-        label: globalStateDeltaTabLabel,
-        children: <GlobalStateDeltas transaction={transaction} />,
-      },
-      {
-        id: localStateDeltaTabId,
-        label: localStateDeltaTabLabel,
-        children: <LocalStateDeltas transaction={transaction} />,
-      },
-    ],
-    [transaction]
-  )
+  const loadableAbiMethod = useAtomValue(loadable(transaction.abiMethod))
 
   return (
     <div className={cn('space-y-2')}>
       <div className={cn('flex items-center justify-between')}>
         <h2>Application Call</h2>
       </div>
-      <DescriptionList items={items} />
-      <Tabs defaultValue={applicationArgsTabId}>
-        <TabsList aria-label={appCallTransactionDetailsLabel}>
-          {tabs.map((tab) => (
-            <TabsTrigger key={tab.id} className="w-44" value={tab.id}>
-              {tab.label}
-            </TabsTrigger>
-          ))}
-        </TabsList>
+      <RenderLoadable loadable={loadableAbiMethod}>
+        {(abiMethod) => (
+          <>
+            <AppCallDescriptionList transaction={transaction} abiMethod={abiMethod} />
+            <AppCallTransactionTabs transaction={transaction} abiMethod={abiMethod} />
+          </>
+        )}
+      </RenderLoadable>
+    </div>
+  )
+}
+
+function AppCallDescriptionList({
+  transaction,
+  abiMethod,
+}: {
+  transaction: AppCallTransaction | InnerAppCallTransaction
+  abiMethod: AbiMethod | undefined
+}) {
+  const items = useMemo(
+    () =>
+      [
+        {
+          dt: transactionSenderLabel,
+          dd: <AccountLink address={transaction.sender} showCopyButton={true} />,
+        },
+        {
+          dt: applicationIdLabel,
+          dd: <ApplicationLink applicationId={transaction.applicationId} showCopyButton={true} />,
+        },
+        ...(abiMethod ? [{ dt: abiMethodNameLabel, dd: abiMethod.name }] : []),
+        {
+          dt: onCompletionLabel,
+          dd: transaction.onCompletion,
+        },
+      ].filter(isDefined),
+    [abiMethod, transaction.applicationId, transaction.onCompletion, transaction.sender]
+  )
+  return <DescriptionList items={items} />
+}
+
+function AppCallTransactionTabs({
+  transaction,
+  abiMethod,
+}: {
+  transaction: AppCallTransaction | InnerAppCallTransaction
+  abiMethod: AbiMethod | undefined
+}) {
+  const tabs = useMemo(
+    () =>
+      [
+        abiMethod
```

Comment:
I changed it to spread operator but to be honest I keep switching because it and return/filter undefined. Do you have any reasons why we should use one over another?